pgx-version-updater/src/main.rs

```diff
@@ -146,90 +160,89 @@ fn main() {
         } else {
-            // Bump pacakge version if we can
-            if doc.contains_key("package") {
-                doc["package"]["version"] = value(args.update_version.clone());
+            // Bump package version if we can
+            if let Some(package_version) = doc.get_mut("package").and_then(|p| p.get_mut("version"))
+            {
+                *package_version = value(args.update_version.clone());
             }
         }
 
         // Process dependencies in each file. Generally dependencies can be found in
         // [dependencies], [dependencies.foo], [build-dependencies], [dev-dependencies]
-        for updatable_table in vec!["dependencies", "build-dependencies", "dev-dependencies"] {
-            if doc.contains_table(updatable_table) {
-                let deps_table: &mut Table = doc
-                    .get_mut(updatable_table)
-                    .unwrap()
-                    .as_table_mut()
-                    .unwrap();
-
-                // Attempt to find auto-extracted package names in the various dependency
-                // declarations
-                for package in &updatable_package_names_set {
-                    if deps_table.contains_key(package) {
-                        let dep_item = deps_table.get_mut(package).unwrap();
-
-                        if dep_item.is_table() {
-                            // Tables can contain other tables, and if that's the case we're
-                            // probably at a case of:
-                            //   [dependencies.pgx]
-                            //   version = "1.2.3"
-                            let old_version = dep_item.get("version").unwrap();
-                            let new_version = parse_new_version(
-                                old_version.as_str().unwrap(),
-                                &args.update_version.as_str(),
-                            );
-                            dep_item["version"] = value(new_version);
-                        } else if dep_item.is_inline_table() {
-                            // Inline table covers the case of:
-                            //   [dependencies]
-                            //   pgx = { version = "1.2.3", features = ["..."] }
-                            let inline_table = dep_item.as_inline_table().unwrap();
-
-                            if inline_table.contains_key("version") {
-                                let old_version = inline_table.get("version").unwrap();
-                                let new_version = parse_new_version(
-                                    old_version.as_str().unwrap(),
+        for updatable_table_name in vec!["dependencies", "build-dependencies", "dev-dependencies"] {
+            if let Some(updatable_table) =
+                doc.get_mut(updatable_table_name).and_then(|i| i.as_table_mut())
+            {
+                for package in &updatable_package_names {
+                    match updatable_table.get_mut(package) {
```

Comment:
Despite finding the simplification of the above `for` loop, I don't feel that would improve this one due to needing considerable control logic anyways. I like the use of the `match` with guards for this structure, since we basically get to use it as a `switch`-like form in this case.